vars/cloudformation.groovy

```diff
@@ -370,10 +392,18 @@ def wait(cf, stackName, successStatus)   {
 }
 
 @NonCPS
-def doesStackExist(cf, stackName) {
+def doesStackExist(cf, stackName, inState = null) {
   try {
     DescribeStacksResult result = cf.describeStacks(new DescribeStacksRequest().withStackName(stackName))
-    return result != null
+    if (result != null) {
+      if (inState == null) {
+        return true
+      } else {
+        return result.getStacks().get(0).getStackStatus() == inState
```

Comment:
If getStacks().len != 1: Should this be reported / logged?

Reply:
That will presumably return an error when `.get(0)` fails.